config/config.go

```diff
@@ -324,6 +325,37 @@ func (cfg BaseConfig) ValidateBasic() error {
 	default:
 		return errors.New("unknown log_format (must be 'plain' or 'json')")
 	}
+
+	switch cfg.ProxyApp {
+	case "kvstore",
+		"kvstore_connsync",
+		"kvstore_unsync",
+		"persistent_kvstore",
+		"persistent_kvstore_connsync",
+		"persistent_kvstore_unsync",
+		"e2e",
+		"e2e_connsync",
+		"e2e_unsync",
+		"noop":
+	default: // network address
```

Comment:
I wonder if we don't have anything on this style inside the `p2p` package. We have to validate addresses there too...

Reply:
90% is done by the `net.Dial` function.